packages/core/src/amazonq/explorer/commonNodes.ts

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+import { Command, DeclaredCommand } from '../../shared/vscode/commands2'
+import { codicon, getIcon } from '../../shared/icons'
+import { telemetry } from '../../shared/telemetry/telemetry'
+import { DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { TreeNode } from '../../shared/treeview/resourceTreeDataProvider'
+
+const localize = nls.loadMessageBundle()
+
+export async function switchToAmazonQ(signIn: boolean = false) {
```

Comment:
We want developers to only use the commands which consume this function and not this function itself.
We should put a warning here to not use this directly